fdbclient/DatabaseConfiguration.cpp

```diff
@@ -224,111 +224,123 @@ bool DatabaseConfiguration::isValid() const {
 StatusObject DatabaseConfiguration::toJSON(bool noPolicies) const {
 	StatusObject result;
 
-	if( initialized ) {
-		std::string tlogInfo = tLogPolicy->info();
-		std::string storageInfo = storagePolicy->info();
-		bool customRedundancy = false;
-		if( tLogWriteAntiQuorum == 0 ) {
-			if( tLogReplicationFactor == 1 && storageTeamSize == 1 ) {
-				result["redundancy_mode"] = "single";
-			} else if( tLogReplicationFactor == 2 && storageTeamSize == 2 ) {
-				result["redundancy_mode"] = "double";
-			} else if( tLogReplicationFactor == 4 && storageTeamSize == 6 && tlogInfo == "dcid^2 x zoneid^2 x 1" && storageInfo == "dcid^3 x zoneid^2 x 1" ) {
-				result["redundancy_mode"] = "three_datacenter";
-			} else if( tLogReplicationFactor == 4 && storageTeamSize == 4 && tlogInfo == "dcid^2 x zoneid^2 x 1" && storageInfo == "dcid^2 x zoneid^2 x 1" ) {
-				result["redundancy_mode"] = "three_datacenter_fallback";
-			} else if( tLogReplicationFactor == 3 && storageTeamSize == 3 ) {
-				result["redundancy_mode"] = "triple";
-			} else if( tLogReplicationFactor == 4 && storageTeamSize == 3 && tlogInfo == "data_hall^2 x zoneid^2 x 1" && storageInfo == "data_hall^3 x 1" ) {
-				result["redundancy_mode"] = "three_data_hall";
-			} else if( tLogReplicationFactor == 4 && storageTeamSize == 2 && tlogInfo == "data_hall^2 x zoneid^2 x 1" && storageInfo == "data_hall^2 x 1" ) {
-				result["redundancy_mode"] = "three_data_hall_fallback";
-			} else {
-				customRedundancy = true;
-			}
+	if (!initialized) {
+		return result;
```

Comment:
This seems to change the behavior. Before this change, `customRedundancy` is set to true and some fields of `result` will be set. Is this change intended?

Reply:
I believe this is not changing the behavior. The whole change I did was just pre-return the `result` if not `initialized`, as the whole function is inside `if (initialized)` clause. The `customRedundancy` test is moved below.